### PR TITLE
STP play selection tests with positions

### PR DESCRIPTION
**Pull request by jordanzliu**
### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [ ] **Start of document comments**: each `.cpp` and `.h` file should have a comment at the start of it. See files in the `thunderbots/software/geom` folder for examples.
- [ ] **Function comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the classes defined in `thunderbots/software/geom`
- [ ] **Remove all commented out code**
- [ ] **Remove extra print statements**: for example, those just used for testing
- [ ] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [ ] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: src/thunderbots/software/test/ai/hl/stp/stp_refbox_game_state_play_selection.cpp [hunk 1]
```diff
@@ -8,6 +8,181 @@
...
+     .friendly_positions = {{-4.5, 0}, {0, 0.0}, {0, 2.5}, {0, -2.5}},
+     .enemy_positions    = {{4.5, 0}, {-(4.5 - 2.0), 2.5}, {-0.3, 0}, {-0.3, -2.5}},
+     .ball_position      = Point(-(4.5 - 1.0), 0),
+     .ball_velocity      = Vector(-1, 0),
+     .first_game_state   = RefboxGameState::PREPARE_PENALTY_THEM,
+     .second_game_state  = RefboxGameState::NORMAL_START}};
+
+// INSTANTIATE_TEST_CASE_P(TestPositions,
+// STPRefboxGameStatePlaySelectionTestWithPositions,
+//                        ::testing::ValuesIn(test_params.begin(), test_params.end()));
```
Comment (garethellis0) [inline src/thunderbots/software/test/ai/hl/stp/stp_refbox_game_state_play_selection.cpp:186]: We should remove commented out code.

File: src/thunderbots/software/test/ai/hl/stp/stp_refbox_game_state_play_selection.cpp [hunk 2]
```diff
@@ -63,12 +239,11 @@ TEST_P(STPRefboxGameStatePlaySelectionTest,
 
 auto all_refbox_game_states = ::Test::TestUtil::getAllRefboxGameStates();
 
-// TODO: uncomment this when plays are completed
 // INSTANTIATE_TEST_CASE_P(AllRefboxGameStates, STPRefboxGameStatePlaySelectionTest,
 //                        ::testing::ValuesIn(all_refbox_game_states.begin(),
-//                        all_refbox_game_states.end()));
+//                                            all_refbox_game_states.end()));
```
Comment (garethellis0) [inline src/thunderbots/software/test/ai/hl/stp/stp_refbox_game_state_play_selection.cpp:248]: Similar comment to above.

File: src/thunderbots/software/test/ai/hl/stp/stp_refbox_game_state_play_selection.cpp [hunk 3]
```diff
@@ -8,6 +8,181 @@
...
+     .ball_position      = Point(-(4.5 - 1.0), 0),
+     .ball_velocity      = Vector(-1, 0),
+     .first_game_state   = RefboxGameState::PREPARE_PENALTY_THEM,
+     .second_game_state  = RefboxGameState::NORMAL_START}};
+
+// INSTANTIATE_TEST_CASE_P(TestPositions,
+// STPRefboxGameStatePlaySelectionTestWithPositions,
+//                        ::testing::ValuesIn(test_params.begin(), test_params.end()));
+
 class STPRefboxGameStatePlaySelectionTest
```
Comment (garethellis0) [inline src/thunderbots/software/test/ai/hl/stp/stp_refbox_game_state_play_selection.cpp:188]: Are we using this class anymore?

File: src/thunderbots/software/test/ai/hl/stp/stp_refbox_game_state_play_selection.cpp [hunk 4]
```diff
@@ -63,12 +239,11 @@ TEST_P(STPRefboxGameStatePlaySelectionTest,
 
 auto all_refbox_game_states = ::Test::TestUtil::getAllRefboxGameStates();
```
Comment (garethellis0) [inline src/thunderbots/software/test/ai/hl/stp/stp_refbox_game_state_play_selection.cpp:244]: Are we using this?

**Review by garethellis0 — CHANGES_REQUESTED** [review]

File: src/thunderbots/software/test/ai/hl/stp/stp_refbox_game_state_play_selection.cpp (hunk 1 above)
Comment (jordanzliu) [reply src/thunderbots/software/test/ai/hl/stp/stp_refbox_game_state_play_selection.cpp:186]: ideally yes, but i wanted to make the change to enable the test very trivial once all plays are in place

File: src/thunderbots/software/test/ai/hl/stp/stp_refbox_game_state_play_selection.cpp (hunk 3 above)
Comment (jordanzliu) [reply src/thunderbots/software/test/ai/hl/stp/stp_refbox_game_state_play_selection.cpp:188]: yes, the test with positions does not go through all states

File: src/thunderbots/software/test/ai/hl/stp/stp_refbox_game_state_play_selection.cpp (hunk 4 above)
Comment (jordanzliu) [reply src/thunderbots/software/test/ai/hl/stp/stp_refbox_game_state_play_selection.cpp:244]: yes

**Review by garethellis0 — APPROVED** [review]
LGTM :+1: :tada:

**Pull request merged**